src/test/java/ut/com/pragbits/bitbucketserver/components/PullRequestActivityListenerTest.java

```diff
@@ -0,0 +1,203 @@
+  }
+
+  @Test
+  public void notifySlackChannelNotInvokedWhenForkedRepoAndOptionDisabled() throws Exception {
+    when(slackSettings.isSlackNotificationsEnabledForPush()).thenReturn(true);
+
+    when(forkedRepository.isFork()).thenReturn(true);
+    when(slackSettings.isSlackNotificationsEnabledForPersonal()).thenReturn(false);
+    when(slackGlobalSettingsService.getSlackNotificationsEnabledForPersonal()).thenReturn(false);
+
+    PullRequestActivityListener listener = new PullRequestActivityListener(slackGlobalSettingsService, slackSettingsService, navBuilder, slackNotifier, avatarService);
+    listener.NotifySlackChannel(event);
+
+    verify(slackNotifier, never()).SendSlackNotification(any(), any());
+  }
+
+  @Test
+  public void notifySlackChannelInvokedWhenEnabled() throws Exception {
+    when(slackSettings.isSlackNotificationsEnabled()).thenReturn(true);
+    when(slackSettings.isSlackNotificationsOverrideEnabled()).thenReturn(true);
+    when(slackSettings.isSlackNotificationsOpenedEnabled()).thenReturn(true);
+
+    PullRequestActivityListener listener = new PullRequestActivityListener(slackGlobalSettingsService, slackSettingsService, navBuilder, slackNotifier, avatarService);
+    listener.NotifySlackChannel(event);
+
+    verify(slackNotifier).SendSlackNotification(eq("https://hooks.slack.com/localhook"), eq(TestFixtures.PR_OVERRIDE_SETTINGS_PAYLOAD));
+  }
+
+  @Test
+  public void notifySlackChannelInvokedWithPatternMatching() throws Exception {
+    when(slackSettings.isSlackNotificationsEnabled()).thenReturn(true);
+    when(slackSettings.isSlackNotificationsOverrideEnabled()).thenReturn(true);
+    when(slackSettings.isSlackNotificationsOpenedEnabled()).thenReturn(true);
+    when(slackSettings.getSlackChannelName()).thenReturn("master/.*->#master_channel,bugfix/.*->#bugfix_channel");
+    when(pullRequestRefTo.getDisplayId()).thenReturn(TestFixtures.BUGFIX_REF_ID);
+
+    PullRequestActivityListener listener = new PullRequestActivityListener(slackGlobalSettingsService, slackSettingsService, navBuilder, slackNotifier, avatarService);
+    listener.NotifySlackChannel(event);
+
+    verify(slackNotifier).SendSlackNotification(eq("https://hooks.slack.com/localhook"), eq(TestFixtures.PR_BUGFIX_CHANNEL_PAYLOAD));
+  }
+
+  @Test
+  public void notifySlackChannelEmptyChannel() throws Exception {
+    when(slackSettings.isSlackNotificationsEnabled()).thenReturn(true);
+    when(slackSettings.isSlackNotificationsOverrideEnabled()).thenReturn(true);
+    when(slackSettings.isSlackNotificationsOpenedEnabled()).thenReturn(true);
+
+    PullRequestActivityListener listener = new PullRequestActivityListener(slackGlobalSettingsService, slackSettingsService, navBuilder, slackNotifier, avatarService);
+    listener.NotifySlackChannel(event);
+
+    verify(slackNotifier).SendSlackNotification(eq("https://hooks.slack.com/localhook"), eq(TestFixtures.PR_NO_CHANNEL_PAYLOAD));
```

Comment:
We should pull out the URL as a constant because it's being reused a couple times and it hasn't changed